### PR TITLE
fix: fix exec failure with karamada-aggregated-apiserver

### PR DESCRIPTION
**Pull request by jwcesign**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Fixes https://github.com/karmada-io/karmada/issues/4197
* Fix exec failure.
* Add karmadctl e2e framework.

**Which issue(s) this PR fixes**:
Part of  #https://github.com/karmada-io/karmada/issues/4197

**Special notes for your reviewer**:
none

**Does this PR introduce a user-facing change?**:

```release-note
`karmada-aggregated-apiserver`: Fix exec failure with karamada-aggregated-apiserver.
```

### CODE REVIEW

**Comment by jwcesign** [comment]
/cc @lonelyCZ @XiShanYongYe-Chang 
Please take a look at the karmadactl e2e framework.

**Comment by jwcesign** [comment]
/cc @XiShanYongYe-Chang 
Now that everything is ready, please take a look.

File: test/e2e/framework/karmadactl.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	KarmadaContext string
+	KarmadactlPath string
+	Namespace      string
+}
+
+func NewKarmadactlCommand(kubeConfig, karmadaContext, karmadactlPath, namespace string, args ...string) *KarmadactlBuilder {
+	builder := new(KarmadactlBuilder)
+	tk := NewTestKubeconfig(kubeConfig, karmadaContext, karmadactlPath, namespace)
+	builder.cmd = tk.KarmadactlCmd(args...)
+	builder.timeout = time.After(10 * time.Second)
```
Comment (XiShanYongYe-Chang) [inline test/e2e/framework/karmadactl.go:48]: Can we configure this timeout?

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks~

File: test/e2e/framework/karmadactl.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+		}
+	case net.Error:
+		if err.Timeout() {
+			return true
+		}
+	}
+	return false
+}
+
+func (k *KarmadactlBuilder) ExecOrDie() (string, error) {
```
[the remark this answers is not in the document]
Comment (jwcesign) [reply test/e2e/framework/karmadactl.go:99]: Please take a look now

File: test/e2e/karmadactl_test.go [hunk 3]
```diff
@@ -557,3 +558,48 @@ var _ = framework.SerialDescribe("Karmadactl cordon/uncordon testing", ginkgo.La
...
 	})
 })
+
+var _ = ginkgo.Describe("Karmadactl exec testing", ginkgo.Labels{NeedCreateCluster}, func() {
+	var policyName string
+	var pod *corev1.Pod
+	var policy *policyv1alpha1.PropagationPolicy
+
+	ginkgo.BeforeEach(func() {
+
```
Comment (XiShanYongYe-Chang) [inline test/e2e/karmadactl_test.go:568]: A blank line.

File: test/e2e/karmadactl_test.go [hunk 4]
```diff
@@ -557,3 +558,47 @@ var _ = framework.SerialDescribe("Karmadactl cordon/uncordon testing", ginkgo.La
...
+	})
+
+	ginkgo.It("Test exec command", func() {
+		framework.WaitPodPresentOnClustersFitWith(framework.ClusterNames(), pod.Namespace, pod.Name,
+			func(pod *corev1.Pod) bool {
+				return pod.Status.Phase != corev1.PodRunning
+			})
+
+		for _, clusterName := range framework.ClusterNames() {
+			cmd := framework.NewKarmadactlCommand(kubeconfig, karmadaContext, karmadactlPath, pod.Namespace, pollTimeout, "exec", pod.Name, "-C", clusterName, "--", "echo", "hello")
```
Comment (XiShanYongYe-Chang) [inline test/e2e/karmadactl_test.go:599]: This timeout is a little long.

**Comment by XiShanYongYe-Chang** [comment]
The newly added e2e is failed.
/lgtm cancel

File: pkg/util/proxy/proxy.go [hunk 5]
```diff
@@ -35,7 +35,7 @@ func ConnectCluster(ctx context.Context, cluster *clusterapis.Cluster, proxyPath
 		return nil, err
 	}
 
-	location, proxyTransport, err := Location(cluster, tlsConfig)
+	location, proxyTransport, err := Location(cluster, tlsConfig.Clone())
```
Comment (RainbowMango) [inline pkg/util/proxy/proxy.go:38]: Can we have a comment to explain why we need to clone the TLSConfig?

Comment (jwcesign) [reply pkg/util/proxy/proxy.go:42]: Added

**Comment by jwcesign** [comment]
/cc @RainbowMango 
Please review again.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**